frontend/ui/translator.lua

```diff
@@ -653,6 +694,31 @@ function Translator:_showTranslation(text, target_lang, source_lang, from_highli
     UIManager:show(textviewer)
 end
 
+function Translator:onTranslateCurrentPage()
+    local ui = require("apps/reader/readerui").instance
+    if not ui or not ui.document then return end
+    local x0, y0, x1, y1, page
+    if ui.rolling then
```

Comment:
It feels all this "reader" business should better be done in your added:
```lua
function ReaderHighlight:onTranslateCurrentPage()
    Translator:onTranslateCurrentPage()
end
```
It is still ReaderHighlight:onTranslateCurrentPage that will always be called, isn't it?

Reply:
Sure, it must be done in ReaderHighlight.